src/imputation_flags.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+
+from src.flag_and_count_matched_pairs import flag_matched_pair_merge
+
+
+def create_impute_flags(df, target, reference, strata, auxiliary):
+    """
+    function to create logical columns for each type of imputation
+    output columns are needed to create the string flag column for
+    imputation methods.
+    Function requires f_predictive and b_predictive columns produced
+    by `flag_matched_pair` function
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        DataFrame containing forward, backward predictive period columns (
+        These columns are created by calling flag_matched_pair_merge forward
+        and backwards)
+
+    target : str
+        column name containing target variable
+    reference : str
+        column name containing business reference id
+    strata : str
+        column name containing strata information (sic)
+    auxiliary : str
+        column name containing auxiliary data
+
+    Returns
+    -------
+    pd.DataFrame
```

Comment:
Perhaps specify that new dataframe returns 4 new bool columns for the flags ? 

This is personal opinion but i feel like if we go to this route with functions and not using a class with constructors and methods we should return a series, if we did then it would more clear what it returns and what the function does